### PR TITLE
Change ScopedVector to VectorRecycler to ease the usage

### PR DESCRIPTION
**Pull request by xiaoxmeng**
Change ScopedVector to VectorRecycler to ease the usage

### CODE REVIEW

File: velox/vector/VectorPool.h [hunk 1]
```diff
@@ -69,19 +69,23 @@ class VectorPool {
 /// NOTE: the context object must provide 'releaseVector' vector such as
 /// EvalCtx.
 template <class Context>
-class ScopedVectorPtr {
+class VectorRecycler {
  public:
-  explicit ScopedVectorPtr(Context& context) : context_(context) {}
-  ~ScopedVectorPtr() {
+  explicit VectorRecycler(VectorPtr& vector, Context& context)
```
Comment (mbasmanova) [inline velox/vector/VectorPool.h:74]: Would it make sense to pas a VectorPool instead of a context?

**Review by mbasmanova — APPROVED** [review]
@xiaoxmeng Looks good to me, but there are CircleCI failures. 

FYI, @laithsakka Laith is looking for even less invasive solution.

**Comment by xiaoxmeng** [comment]
> Laith is looking for even less invasive solution.

Is it to overload the deleter of the shared ptr? It is a very good solution. But I am not very sure if we can guarantee the associated vector pool is always valid when the user drop the last shared reference on the buffer allocated from the vector pool?

File: velox/vector/VectorPool.h (hunk 1 above)
Comment (xiaoxmeng) [reply velox/vector/VectorPool.h:74]: Yeah, good point!

**Review by mbasmanova — APPROVED** [review]

**Comment by mbasmanova** [comment]
> > Laith is looking for even less invasive solution.
> 
> Is it to overload the deleter of the shared ptr? It is a very good solution. But I am not very sure if we can guarantee the associated vector pool is always valid when the user drop the last shared reference on the buffer allocated from the vector pool?

Yes, my understanding is this is the approach Laith is looking into. CC: @laithsakka

**Pull request closed without merging**